internal/pkg/cli/pipeline_status_test.go

```diff
@@ -8,55 +8,356 @@ import (
+					mocks.pipelineSvc.EXPECT().ListPipelineNamesByTags(testTags).Return(nil, mockErr),
+				)
+			},
+			expectedErr: fmt.Errorf("list pipelines: %w", mockErr),
+		},
+		"wraps error when no pipelines selected": {
+			testAppName: mockApplicationName,
+			setupMocks: func(mocks pipelineStatusMocks) {
+				gomock.InOrder(
+					mocks.ws.EXPECT().ReadPipelineManifest().Return(nil, workspace.ErrNoPipelineInWorkspace),
+					mocks.pipelineSvc.EXPECT().ListPipelineNamesByTags(testTags).Return(mockPipelines, nil),
+					mocks.prompt.EXPECT().SelectOne(fmt.Sprintf(fmtPipelineStatusPipelineNamePrompt, color.HighlightUserInput(mockApplicationName)), pipelineStatusPipelineNameHelpPrompt, mockPipelines).Return("", mockErr),
+				)
+			},
+			expectedApp:      mockApplicationName,
+			expectedPipeline: mockPLName,
+			expectedErr:      fmt.Errorf("select pipeline for application %s: %w", mockApplicationName, mockErr),
+		},
+		"success with flags": {
+			testAppName:      mockApplicationName,
+			testPipelineName: mockPLName,
+
+			setupMocks:       func(mocks pipelineStatusMocks) {},
+			expectedApp:      mockApplicationName,
+			expectedPipeline: mockPLName,
+			expectedErr:      nil,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := mocks.NewMockstore(ctrl)
+			mockWS := mocks.NewMockwsPipelineReader(ctrl)
+			mockPrompt := mocks.NewMockprompter(ctrl)
+			mockPLSvc := mocks.NewMockpipelineStateGetter(ctrl)
+			mockSel := mocks.NewMockappSelector(ctrl)
+
+			mocks := pipelineStatusMocks{
+				store:       mockStore,
+				ws:          mockWS,
+				prompt:      mockPrompt,
+				pipelineSvc: mockPLSvc,
+				sel:         mockSel,
+			}
+
+			tc.setupMocks(mocks)
+
+			opts := &pipelineStatusOpts{
+				pipelineStatusVars: pipelineStatusVars{
+					GlobalOpts: &GlobalOpts{
+						prompt:  mockPrompt,
+						appName: tc.testAppName,
+					},
+					pipelineName: tc.testPipelineName,
+				},
+				store:       mockStore,
+				ws:          mockWS,
+				pipelineSvc: mockPLSvc,
+				sel:         mockSel,
+			}
+
+			// WHEN
+			err := opts.Ask()
+
+			// THEN
+			if tc.expectedErr != nil {
+				require.EqualError(t, err, tc.expectedErr.Error())
+			} else {
+				require.Nil(t, err)
+				require.Equal(t, tc.expectedApp, opts.AppName(), "expected application names to match")
+				require.Equal(t, tc.expectedPipeline, opts.pipelineName, "expected pipeline name to match")
+			}
+		})
+	}
+}
+
 func TestPipelineStatus_Execute(t *testing.T) {
-	mockError := errors.New("some error")
+	mockPipelineStatus := mockDescribeData{
+		data: "mockData",
+		err:  mockErr,
+	}
 	testCases := map[string]struct {
-		shouldOutputJSON    bool
-		pipelineName        string
-		mockStatusDescriber func(m *mocks.MockpipelineStatusDescriber)
-		wantedError         error
+		shouldOutputJSON bool
+		pipelineName     string
+		setupMocks       func(m *mocks.Mockdescriber)
```

Comment:
Any reason not to add the mockDescriber to `pipelineStatusMocks`? It might be good just for consistency.